freud/box.pyx

```diff
@@ -341,6 +341,59 @@ cdef class Box:
 
         return np.squeeze(vecs) if flatten else vecs
 
+    def center_of_mass(self, vecs, masses=None):
+        R"""Compute center of mass of an array of vectors, using periodic boundaries.
+
+        Args:
+            vecs (:math:`\left(N, 3\right)` :class:`numpy.ndarray`):
+                Vectors used to find center of mass.
+            masses (:math:`\left(N, 3\right)` :class:`numpy.ndarray`):
+                Masses corresponding to each vector, defaulting to 1 if not
+                provided or :code:`None` (Default value = :code:`None`).
+
+        Returns:
+            :math:`\left(3\right)` :class:`numpy.ndarray`:
+                Center of mass.
+        """  # noqa: E501
+        vecs = freud.util._convert_array(vecs, shape=(None, 3))
+        if masses is None:
+            masses = np.ones(len(vecs), dtype=np.float32)
+        masses = freud.util._convert_array(masses, shape=(len(vecs), ))
```

Comment:
Why explicitly handle `masses==None` here instead of relaying to the C++ function?

Reply:
Good idea. It's a little awkward to use `NULL` in Cython, but I agreed with you and changed the behavior to use `NULL` because I've used the same behavior in other places (e.g. cluster keys).